File: C#/MaximumSubarray.cs
public class Solution53 {
    public int MaxSubArray(int[] nums) {
        int res = nums[0];
        int curr = nums[0];
        for(int i = 1; i < nums.Length; i++) {
            curr += nums[i];
            if (curr < 0 || nums[i] > curr)
                curr = nums[i];
            if (res < curr)
                res = curr;
        }
        return res;  
    }
}

File: Golang/Maximum Subarray.go
func maxSubArray(nums []int) int {
    res := nums[0]
    total := 0
    for _, n := range nums {
        total = total + n
        res = max(res, total)
        if total < 0 {
            total = 0
        }
    }
    return res
}

func max(a int, b int) int {
    if a < b {
        return b
    }
    return a
}